### DISCUSSION TITLE
Which iPhone applications support UTLS? کدام برنامه های آیفون از یو تی ال اس پشتیبانی میکنن؟

### DISCUSSION THREAD

**Discussion by R30na** [hiddify/Hiddify-Manager#173] [Q&A سوالات رایج] [answered]
سلام بعد از اضافه شدن قابلیت uTLS  من چند تا کانفیگ روی napsternetV تست کردم انگار این تنظیمات رو اعمال نمیکنه. 
میخواستم ببینم اگه مثلا بعد از اضافه کردن تنظیمات دستی این تنظیمات رو توی json اضافه کنیم اعمال میشه؟ 
و اینکه برنامه دیگه ای برای آیفون سراغ دارین که این تنظیمات رو از روی url تشخیص بده؟

**Comment by robin98** [comment, marked as answer]
وقتی که کانفیگ رو با فرمت json به nV ایمپورت می کنیم و در همون json، مقادیر uTLS و alpn وجود داشته باشند اپلیکیشن nV بدون هیچ اروری اون رو ایمپورت می کنه و میشه باهاش وصل شد، اما این به هیچ عنوان ثابت نمی کنه که nV واقعا از این قوانین تبعیت می کنه یا خیر.
در واقع بهترین کار این هست که از سمت سرور، سرویس دهنده جوری کانفیگ بشه که uTLS و alpn خاصی رو فقط اجازه بده و تازه با این شرایط هست که میتونیم مطمئن باشیم چه کلاینتی واقعا ساپورت میکنه و چه کلاینتی فقط واسه دکور داره اون مقادیر رو نشون میده.
البته در هایدیفای نمیشه همچین کاری کرد، ولی ممکنه در x-ui روشی برای اینکار وجود داشته باشه.

راه دیگه مانیتور کردن پکت های خروجی دستگاه iOS هست و اینکه ببینید مقدار SNI قابل مشاهده هست یا اینکه انکریپت شده؟ (احتمالا با وایرشارک شدنی باشه)

در برنامه های رایگان من مورد خاصی رو ندیدم که پشتیبانی کامل و بروزی داشته باشند، شاید برنامه های پولی طور دیگری باشند که نیازمند این هست دوستانی که تجربه کار باهاشون رو دارند نظر بدهند.

**Reply by R30na** [reply]
یه سوال دیگه اینکه توی تنظیمات uTLS مهمه که همه مقادیر وارد بشن؟ بعضی کانفیگ ها دیدم فقط مثلا fingerprint وارد میکنن و alpn  رو ست نمیکنن.

**Reply by hiddify-com** [reply]
متاسفانه خود xray گفته نمیتونیم تشخیص بدیم

**Reply by R30na** [reply]
آیا هایدیفای از xray-core استفاده میکنه ؟

**Reply by hiddify-com** [reply]
بله

**Comment by robin98** [comment]
پس از تهیه اپلیکیشن stash متوجه شدم اپلیکیشن جامع و نسبتا خوبی هستش و امکان شخصی سازی و کانفیگ نویسی فوق العاده ای رو به کاربر میده. دقیقا مشابه پروژه https://github.com/Fndroid/clash_for_windows_pkg
از اینجا میتونید stash رو ببینید https://apps.apple.com/us/app/stash-rule-based-proxy/id1596063349
در مورد پشتیبانی از میکرو تی ال اس یا uTLS هم ظاهرا با مقداری به شکل زیر در فایل های کانفیگ از اون پشتیبانی می کنه:
```
    client-fingerprint: chrome   
    #  可选："chrome","firefox","safari","ios","random","none" options.
    #  Utls is currently support TLS transport in TCP/grpc/WS/HTTP for VLESS/Vmess and trojan.
```

**Reply by hiddify-com** [reply]
گویا napsterv هم اضافه کرد
